x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -102,6 +103,51 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
+  async rolloverAliasIfNeeded({ namespace }: { namespace?: string }) {
+    const clusterClient = await this.getClusterClient();
+    const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    const { body: simulatedRollover } = await clusterClient.indices.rollover({
```

Comment:
Why do we need to simulate a rollover? Can't we get the current write index, simulate the index template, and compare the two? (And I think those can happen in parallel)

Reply:
We have to make sequential calls either way - one call to retrieve the concrete index name for the alias write index, then a second set of calls to get the mapping for the concrete index and the simulated mapping for the new index. If we do it with the simulated rollover, elasticsearch handles extracting the current write index and the new index name for us and makes them available as old_index and new_index. If we avoid using simulated rollover, then we have to manually fetch the alias and extract the write index and also manually compute the next index name. So we don't necessarily need to simulate a rollover, but it's simpler this way.

Reply:
@marshallmain could you please add this clarification to the code as a comment?

Reply:
Added explanatory comment in https://github.com/elastic/kibana/pull/102586/commits/a45a70a842416b45d1be340877043f1fd48092fe